examples/repeated-compute/src/main.rs

```diff
@@ -0,0 +1,248 @@
+        log::info!("Results: {printed_numbers:?}");
+    }
+}
+
+fn generate_rand() -> u16 {
+    let mut bytes = [0u8; 2];
+    getrandom::getrandom(&mut bytes[..]).unwrap();
+    u16::from_le_bytes(bytes)
+}
+
+async fn compute(local_buffer: &mut [u32], context: &WgpuContext) {
+    log::info!("Beginning GPU compute on data {local_buffer:?}.");
+    // Local buffer contents -> GPU storage buffer
+    /* Adds a write buffer command to the queue. This command is more complicated
+    than it appears. */
+    context.queue.write_buffer(
+        &context.storage_buffer,
+        0,
+        bytemuck::cast_slice(local_buffer),
+    );
+    log::info!("Wrote to buffer.");
+
+    let mut command_encoder = context
+        .device
+        .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+
+    {
+        let mut compute_pass =
+            command_encoder.begin_compute_pass(&wgpu::ComputePassDescriptor { label: None });
+        compute_pass.set_pipeline(&context.pipeline);
+        compute_pass.set_bind_group(0, &context.bind_group, &[]);
+        compute_pass.dispatch_workgroups(local_buffer.len() as u32, 1, 1);
+    }
+    // We finish the compute pass by dropping it.
+
+    // Entire storage buffer -> staging buffer.
+    command_encoder.copy_buffer_to_buffer(
+        &context.storage_buffer,
+        0,
+        &context.output_staging_buffer,
+        0,
+        context.storage_buffer.size(),
+    );
+
+    // Finalize the command encoder, add the contained commands to the queue and flush.
+    context.queue.submit(Some(command_encoder.finish()));
+    log::info!("Submitted commands.");
+
+    // Finally time to get our results.
+    /* First we get a buffer slice which represents a chunk of the buffer (which we
+    can't access yet). */
+    // We want the whole thing so use unbounded range.
+    let buffer_slice = context.output_staging_buffer.slice(..);
+    /* Now things get complicated. WebGPU, for safety reasons, only allows either the GPU
+    or CPU to access a buffer's contents at a time. We need to "map" the buffer which means
+    flipping ownership of the buffer over to the CPU and making access legal. We do this
+    with `BufferSlice::map_async`. The problem is that map_async is not an async function
+    and we can't await it. What we need to do instead is pass in a closure that will
+    be executed when the slice is either mapped or the mapping has failed. The problem
+    with this is that we don't have a reliable way to wait in the main code for the buffer
+    to be mapped and even worse, calling get_mapped_range or get_mapped_range_mut prematurely
+    will cause a panic, not return an error. Using async channels solves this as awaiting
+    the recieving of a message from the passed closure will force the outside code to wait.
+    It also doesn't hurt if the closure finishes before the outside code catches up as
+    the message is buffered and recieving will just pick that up. */
```

Comment:
Really minor nit: we should use // comments everywhere and maybe break this up into smaller paragraphs.

Just feels a bit "wall of text"-ey

Reply:
Sure. See if you like the way I broke it up.

Reply:
Oh this is in https://github.com/gfx-rs/wgpu/pull/3885/commits/42586901e9efcab52d465e3c40b557a6fd17fe5c btw just so it's easy to find.